File: Chapter_3/project_3_2.pl
use strict;
use warnings;

use Time::Piece;

print "Enter item number: ";
chomp( my $item_number = <STDIN> );

print "Enter unit price: ";
chomp( my $unit_price = <STDIN> );

print "Enter purchase date (mm/dd/yyyy): ";
chomp( my $purchase_date = <STDIN> );

my $tp = Time::Piece->strptime( $purchase_date, '%m/%d/%Y' );

my $date = $tp->strftime('%m/%d/%Y');

print "\n";
print "Item\t Unit\t Purchase\n\t Price\t Date\n";

printf sprintf( "%-3d\t\$%.2f\t%s\n", $item_number, $unit_price, $date );
printf("_______________________________________\n");

File: Chapter_3/project_3_2.py
from datetime import datetime
from beautifultable import BeautifulTable

order_number = int(input("Please enter the order number: "))
unit_price = float(input("Please enter the unit price: "))
date = input("Please enter the date mm/dd/yyyy: ")
dt = datetime.strptime(date, '%m/%d/%Y')
my_date = f"{dt.month}/{dt.day}/{dt.year}"

print("\nItem\t Unit\t Purchase\n\t Price\t Date\n")
print(f"{order_number}\t${unit_price}\t {my_date}")
print("_________________________________________")

table = BeautifulTable()
table.column_headers = ["Item Number", "Unit Price", "Purchase Date"]
table.append_row([order_number, unit_price, my_date])
print(table)
